contracts/Lotto.sol

```diff
@@ -0,0 +1,287 @@
+        uint256 costPerTicket,
+        uint256 startingBlock,
+        uint256 closingBlock,
+        uint256 endBlock,
+        address indexed creator
+    );
+
+    event newBatchMint(
+        address indexed minter,
+        uint256[] ticketIDs,
+        uint32[] numbers,
+        uint256 totalCost,
+        uint256 discount,
+        uint256 pricePaid
+    );
+
+    //-------------------------------------------------------------------------
+    // MODIFIERS
+    //-------------------------------------------------------------------------
+
+
+
+    //-------------------------------------------------------------------------
+    // CONSTRUCTOR
+    //-------------------------------------------------------------------------
+
+    constructor(
+        address _cake, 
+        uint8 _sizeOfLotteryNumbers,
+        uint8 _maxValidNumberRange    
+    ) {
+        cake_ = IERC20(_cake);
+        sizeOfLottery_ = _sizeOfLotteryNumbers;
+        maxValidRange_ = _maxValidNumberRange;
+    }
+
+    function init(address _lotteryNFT) public onlyOwner() {
+        nft_ = ILottoNFT(_lotteryNFT);
+    }
+
+    //-------------------------------------------------------------------------
+    // VIEW FUNCTIONS
+    //-------------------------------------------------------------------------
+
+    /**
+      * @return Gets the current time
+      */
+    function getTime() public view returns(uint256) {
+        return block.timestamp;
+    }
+
+    function costToBuyTickets(
+        uint256 _lotteryID,
+        uint256 _numberOfTickets
+    ) 
+        public 
+        view 
+        returns(uint256 totalCost) 
+    {
+        uint256 pricePer = allLotteries_[_lotteryID].costPerTicket;
+        totalCost = pricePer.mul(_numberOfTickets);
+        // TODO use internal bonding curve 
+    }
+
+    function getBasicLottoInfo(uint256 _lotteryID) public view returns(
+        LottoInfo memory
+    )
+    {
+        return(
+            allLotteries_[_lotteryID]
+        ); 
+    }
+
+    //-------------------------------------------------------------------------
+    // STATE MODIFYING FUNCTIONS 
+    //-------------------------------------------------------------------------
+
+    //-------------------------------------------------------------------------
+    // Restricted Access Functions
+
+    function updateSizeOfLottery(uint8 _newSize) public onlyOwner() {
+        require(
+            sizeOfLottery_ != _newSize,
+            "Cannot set to current size"
+        );
+        sizeOfLottery_ = _newSize;
+    }
+
+    function updateMaxRange(uint8 _newMaxRange) public onlyOwner() {
+        require(
+            maxValidRange_ != _newMaxRange,
+            "Cannot set to current size"
+        );
+        maxValidRange_ = _newMaxRange;
+    }
+
+
+    function drawWinningNumbers(uint256 _lottoID) public onlyOwner() {
+        // Creating space for winning numbers
+        uint8[] memory winningNumbers;
+        // TODO will call ChainLink VRF
```

Comment:
?

Reply:
Waiting on the green from Chef Hops for ChainLink integration BSC. 

In the meantime going to make a mock contract + interface